docs/scripting-element-object.rst

```diff
@@ -0,0 +1,393 @@
+call; ``method`` provides an information about error type; otherwise ``value`` is a function
+that can be called from Lua to execute node method in page context.
+
+**Async:** no.
+
+See more in :ref:`splash-jsfunc`.
+
+.. note::
+
+    If the specified method returns another HTML element it will be represented as a table with
+    ``type`` property which equals to ``'node'``
+
+
+.. _splash-element-mouse-click:
+
+element:mouse_click
+-------------------
+
+Trigger mouse click event on the top-left corner of the element.
+
+**Signature:** ``ok, reason = element:mouse_click()``
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+See more about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-element-mouse-hover:
+
+element:mouse_click
+-------------------
+
+Trigger mouse hover (JavaScript mouseover) event on the top-left corner of the element.
+
+**Signature:** ``ok, reason = element:mouse_hover()``
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+See more about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-element-get-styles:
+
+element:get_styles
+------------------
+
+Return the computed styles of the element.
+
+**Signature:** ``ok, styles = element:get_styles()``
+
+**Returns:** ``ok, styles`` pair. If ``ok`` is nil then error happened during the
+function call; ``styles`` provides an information about error type; otherwise
+``styles`` is a table with computed styles of the element.
+
+**Async:** no.
+
+Example of getting the font size of the element.
+
+.. code-block:: lua
+
+    function main(splash)
+        local element = splash:select('.element')
+        local ok, styles = assert(element:get_styles())
+
+        return styles['font-size']
+    end
+
+
+.. _splash-element-get-bounds:
+
+element:get_bounds
+------------------
+
+Return the bounding client rectangle of the element
+
+**Signature:** ``ok, styles = element:get_bounds()``
+
+**Returns:** ``ok, bounds`` pair. If ``ok`` is nil then error happened during the
+function call; ``bounds`` provides an information about error type; otherwise
+``bounds`` is a table with the client bounding rectangle with the ``top``, ``right``,
+``bottom`` and ``left`` coordinates.
+
+**Async:** no.
+
+Example of getting the bounds of the element.
+
+.. code-block:: lua
+
+    function main(splash)
+        local element = splash:select('.element')
+        local ok, bounds = assert(element:get_bounds())
+        -- e.g. bounds is { top = 10, right = 20, bottom = 20, left = 10 }
+        return bounds
+    end
+
+
+.. _splash-element-png:
+
+element:png
+-----------
+
+Return a screenshot of the element in PNG format
+
+**Signature:** ``ok, shot = element:png{width=nil, height=nil, scale_method='raster'}``
```

Comment:
What about making the "padding" relative coordinates, i.e. positive values mean normal html padding, negative values - pad the image "inside" of the element.

On a sidenote, i don't think it's necessary to keep "width" and "height"  parameters with the same, quite obscure meaning as in the api. We might want to seize the opportunity to think of better names or better ways to describe the use cases for these parameters, not necessarily within this project.